examples/extra_element_integer/clustering_algorithm.h

```diff
@@ -120,10 +124,16 @@ const unsigned int FindCluster(libMesh::Mesh &mesh,
                         /*I have to pass the parent element here cause there is a big chance when doing it recursively
```

Comment:
One solution is to initialize the cluster ID to be something like -1, and use that to determine whether an element has been visited.  It may also be more efficient because you can do it when creating the extra_element_integer